### DISCUSSION TITLE
Command not found - installing ganache-cli with yarn on Visual Studio

### DISCUSSION THREAD

**Discussion by darkmansaturne** [smartcontractkit/full-blockchain-solidity-course-py#707] [Q&A] [answered]
I've installed nodeJS in the terminal of Visual Studio version :
```v16.13.1```

Yarn
```1.22.17```

Ganache-cli

`MacBook:web3_py_simple_storage myName$ yarn global add ganache-cli`

`yarn global v1.22.17
warning ../package.json: No license field
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Installed "ganache-cli@6.12.2" with binaries:
      - ganache-cli
✨  Done in 1.10s. `

BUT when I try to do `ganache-cli --version`, I always received the same msg `bash: ganache-cli: command not found`... I maybe think it's a path problem but I tried a lot and a lot of solution and still nothing.. 

I really thanks a lot in advance the guy who will help me !

**Comment by cromewar** [comment, marked as answer]
Hello @darkmansaturne 
Yep this is due to the node_modules are not added to your path ( in this case the packages you add with yarn), i strongly suggest you to reinstall node using brew (is the best option for mac), and also get a look about how to add node packages to path on Macos (depends a lot on the version so you would have to do a little bit of googling).

**Reply by darkmansaturne** [reply]
Thank you for responding ! That's what I done but look, I have ganache-cli installed in my .bin directory. I use the right path to use ganache-cli and still `command not found`...

<img width="225" alt="Capture d’écran 2022-01-06 à 11 43 22" src="https://user-images.githubusercontent.com/97000651/148370852-c2115a29-cbe4-4da9-8246-e0045a506bbe.png">

`MacBook:web3_py_simple_storage myname$ ganache-cli --version
bash: ganache-cli: command not found
MacBook:web3_py_simple_storage myname$ ganache-cli
bash: ganache-cli: command not found
MacBook:web3_py_simple_storage myname$ yarn --version
1.22.17
MacBook:web3_py_simple_storage myname$ node --version
v16.13.1
MacBook:web3_py_simple_storage myname$ ganache --version
bash: ganache: command not found
MacBook:web3_py_simple_storage myname$ cd /Users/myname/demos/web3_py_simple_storage/node_modules/.bin
MacBook:.bin myname$ use ganache-cli
bash: use: command not found
MacBook:.bin myname$ ganache-cli
bash: ganache-cli: command not found
MacBook:.bin myname$ find . -name ganache-cli
./ganache-cli`

**Reply by darkmansaturne** [reply]
Okay I find it.. I'm a big shit 😃 If people have the same problem, just write `yarn ganache-cli` instead of `ganache-cli` ! Thanks @cromewar 🤘

**Reply by cromewar** [reply]
That's why a suggested to add yarn packages to path, to avoid using yarn each time. But I'm glad you find this workaround.

**Reply by shucks18** [reply]
haha same issue thanks @darkmansaturne
